libs/community/langchain_community/utils/vertex_functions.py

```diff
@@ -0,0 +1,41 @@
+from typing import TYPE_CHECKING
+
+from langchain_core.tools import Tool
+
+from langchain_community.utils.openai_functions import (
+    FunctionDescription,
+    convert_pydantic_to_openai_function,
+)
+
+if TYPE_CHECKING:
+    from vertexai.preview.generative_models import Tool as VertexTool
+
+
+def format_tool_to_vertex_function(tool: Tool) -> FunctionDescription:
+    "Format tool into the Vertex function API."
+    if tool.args_schema:
+        return convert_pydantic_to_openai_function(
```

Comment:
i tried this out and it errors, i think you need to remove `title` from the json schema that is generated

Reply:
@hwchase17 which error do you get?

I've just tried this again
```
raw_tools = [
    Tool(
        name="Search",
        func=search.run,
        description="useful for when you need to answer questions about current events. You should ask targeted questions",
    ),
    Tool(
        name="Calculator",
        func=llm_math_chain.run,
        description="useful for when you need to answer questions about math",
    )
]
tools = [format_tool_to_vertex_tool(tool) for tool in raw_tools]
```

and it seems to be working fine.

and there's also a working integration test: 
https://github.com/langchain-ai/langchain/blob/45f2fd5a70b8df54a919c47dc16b0a02c87b0f15/libs/community/tests/integration_tests/chat_models/test_vertexai.py#L299